src/deploy/firestore/deploy.ts

```diff
@@ -45,22 +76,33 @@ async function deployIndexes(context: any, options: any): Promise<void> {
       }
       const fieldOverrides = indexesRawSpec.fieldOverrides || [];
 
-      await firestoreIndexes.deploy(options, indexes, fieldOverrides, databaseId).then(() => {
-        utils.logSuccess(
-          `${clc.bold(clc.green("firestore:"))} deployed indexes in ${clc.bold(
-            indexesFileName,
-          )} successfully for ${databaseId} database`,
-        );
-      });
+      try {
+        await firestoreIndexes.deploy(options, indexes, fieldOverrides, databaseId);
+      } catch (err: any) {
+        if (err.status !== 404) {
+          throw err;
+        }
+        // It might take a while for the database to be created.
```

Comment:
Is there a cleaner way to handle this case rather than justign waiting? Is there an operation that we ought to be polling on?

Can be done in a follow up if its annoyingly complex.